plugins/devlink/fu-devlink-netlink.h

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright 2025 NVIDIA Corporation & Affiliates
+ *
+ * SPDX-License-Identifier: LGPL-2.1-or-later
+ */
+
+#pragma once
+
+#include <fwupdplugin.h>
+
+#include <libmnl/libmnl.h>
+#include <linux/devlink.h>
+#include <linux/genetlink.h>
+#include <linux/netlink.h>
+
+/* generic netlink socket wrapper */
+typedef struct {
+	struct mnl_socket *nl;
+	gchar *buf;
+	guint32 family_id;
+	guint32 config_group_id;
+	FuDevice *device; /* device for emulation recording/playback */
+	gboolean is_emulated;
+	gboolean save_events;
+	gint pipe_fds[2];  /* dummy pipe for emulation (read=0, write=1) */
+} FuDevlinkGenSocket;
+
+#define FU_DEVLINK_NETLINK_BUF_SIZE MNL_SOCKET_BUFFER_SIZE
+
+/* send/receive functions */
+gboolean
+fu_devlink_netlink_msg_run(FuDevlinkGenSocket *nlg,
+			   gsize len,
+			   guint32 seq,
+			   mnl_cb_t cb,
+			   void *data,
+			   GError **error) G_GNUC_NON_NULL(1, 4);
+
+gboolean
+fu_devlink_netlink_msg_send_recv(FuDevlinkGenSocket *nlg,
+				 struct nlmsghdr *nlh,
+				 mnl_cb_t cb,
+				 void *data,
```

Comment:
i think we're using `gpointer userdata` rather than `void *data` in other places